src/encode.rs

```diff
@@ -0,0 +1,296 @@
+    /// ```
+    pub fn to_pem_pkcs1(&self) -> Result<String> {
+        let pem = pem::Pem {
+            tag: String::from("RSA PRIVATE KEY"),
+            contents: self.to_pkcs1()?,
+        };
+        Ok(pem::encode(&pem))
+    }
+
+    /// Converts RSA Private key into `PKCS8` encoded bytes in pem format.
+    ///
+    /// Encodes the key with the header:
+    /// `-----BEGIN PRIVATE KEY-----`
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use rsa::RSAPrivateKey;
+    /// use rand::rngs::OsRng;
+    ///
+    /// let mut rng = OsRng;
+    /// let bits = 2048;
+    /// let private_key = RSAPrivateKey::new(&mut rng, bits).expect("failed to generate a key");
+    ///
+    /// let _ = private_key.to_pem_pkcs8();
+    /// ```
+    pub fn to_pem_pkcs8(&self) -> Result<String> {
+        let pem = pem::Pem {
+            tag: String::from("PRIVATE KEY"),
+            contents: self.to_pkcs8()?,
+        };
+        Ok(pem::encode(&pem))
+    }
+}
+
+#[cfg(feature = "pem")]
+impl RSAPublicKey {
+    /// Converts RSA Public key into `PKCS1` encoded bytes in pem format.
+    ///
+    /// Encodes the key with the header:
+    /// `-----BEGIN RSA PUBLIC KEY-----`
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use rsa::{RSAPrivateKey, RSAPublicKey};
+    /// use rand::rngs::OsRng;
+    ///
+    /// let mut rng = OsRng;
+    /// let bits = 2048;
+    /// let private_key = RSAPrivateKey::new(&mut rng, bits).expect("failed to generate a key");
+    /// let public_key = RSAPublicKey::from(&private_key);
+    ///
+    /// let _ = public_key.to_pem_pkcs1();
+    /// ```
+    pub fn to_pem_pkcs1(&self) -> Result<String> {
+        let pem = pem::Pem {
+            tag: String::from("RSA PUBLIC KEY"),
+            contents: self.to_pkcs1()?,
+        };
+        Ok(pem::encode(&pem))
+    }
+    /// Converts RSA Public key into `PKCS8` encoded bytes in pem format.
+    ///
+    /// Encodes the key with the header:
+    /// `-----BEGIN PUBLIC KEY-----`
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use rsa::{RSAPrivateKey, RSAPublicKey};
+    /// use rand::rngs::OsRng;
+    ///
+    /// let mut rng = OsRng;
+    /// let bits = 2048;
+    /// let private_key = RSAPrivateKey::new(&mut rng, bits).expect("failed to generate a key");
+    /// let public_key = RSAPublicKey::from(&private_key);
+    ///
+    /// let _ = public_key.to_pem_pkcs8();
+    /// ```
+    pub fn to_pem_pkcs8(&self) -> Result<String> {
+        let pem = pem::Pem {
+            tag: String::from("PUBLIC KEY"),
+            contents: self.to_pkcs8()?,
+        };
+        Ok(pem::encode(&pem))
+    }
+}
+
+fn to_bigint(value: &crate::BigUint) -> simple_asn1::BigInt {
+    // TODO can be switched if simple_asn1 BigInt type is updated
+    // This is not very clean because of the exports available from simple_asn1
+    simple_asn1::BigInt::from_signed_bytes_le(&value.to_bigint().unwrap().to_signed_bytes_le())
+}
+
+impl RSAPrivateKey {
```

Comment:
For `PrivateKey` implementations, we can't reliably access their private components (which may be inaccessible e.g. for YubiKeys). Instead, let's have a `trait PrivateKeyEncoding : PrivateKey` that defines the `to_pkcs1` and `to_pkcs8` interfaces, and then this would be:
```suggestion
impl PrivateKeyEncoding for RSAPrivateKey {
```